### PR TITLE
More Mexican foods (real)

### PR DESCRIPTION
**Pull request by onura46**
#### Summary
Content "More Mexican foods"

#### Purpose of change
Adds three new Mexican foods, tweaks item spawns, tweaks recipes to allow new items to be used.

#### Describe the solution
Adds three new foods with recipes learnable through the Mexican cookbook:
- Flour tortillas
- Salsa
- Horchata (cannot be autolearned)

These items have been added to item spawn groups in kitchens, restaurants, fridges, and pantries where appropriate. A new food requirement called `tortilla_any` has been created so you can now use either type of tortilla for some recipes!

This PR also removes `autolearn` from hominy, which I intended to do in the initial PR. Hominy is made with lye or slaked lime using a chemical process called nixtamalization to loosen the hull and treat the kernel. For most people, this is not a colloquially-known process because it involves real chemistry knowledge and non-standard ingredients. Perhaps because of this or because of how long it takes to make, hominy is normally store-bought rather than made at home.

This is effectively half an implementation of #61697. I have decided that elote and gorditas are just too similar to existing foods to be meaningfully different, barely off from "corn on the cob" and "deluxe sandwich", respectively. Though the real-world differences are obvious, the difference in-game is negligible.

#### Testing
Many spawns and repeated testing to make sure recipes were balanced correctly and everything spawned as expected.

#### Additional context
No more stale draft PRs for me lmao.

### CODE REVIEW

File: data/json/items/comestibles/bread.json [hunk 1]
```diff
@@ -400,6 +400,28 @@
...
+    "type": "COMESTIBLE",
+    "id": "tortilla_flour",
+    "name": { "str": "flour tortilla" },
+    "weight": "24 g",
+    "color": "brown",
+    "spoils_in": "15 days",
+    "container": "bag_plastic",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 64,
```
Comment (Maleclypse) [inline data/json/items/comestibles/bread.json:413]:
```suggestion
    "calories": 100,
```

File: data/json/items/comestibles/drink.json [hunk 2]
```diff
@@ -598,6 +598,32 @@
...
+    "price_postapoc": 75,
+    "phase": "liquid",
+    "container": "bottle_plastic",
+    "sealed": false,
+    "material": [ "milk", "junk" ],
+    "primary_material": "milk",
+    "symbol": "~",
+    "color": "brown",
+    "comestible_type": "DRINK",
+    "calories": 200,
```
Comment (Maleclypse) [inline data/json/items/comestibles/drink.json:620]:
```suggestion
    "calories": 225,
```

File: data/json/items/comestibles/drink.json [hunk 3]
```diff
@@ -598,6 +598,32 @@
...
     "flags": [ "EATEN_HOT" ]
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "horchata",
+    "looks_like": "milk",
+    "category": "food",
+    "name": { "str_sp": "horchata" },
+    "description": "A sugary drink from Central America infused with rice, milk, and cinnamon.",
+    "weight": "257 g",
```
Comment (Maleclypse) [inline data/json/items/comestibles/drink.json:608]:
```suggestion
    "weight": "237 g",
```

File: data/json/items/comestibles/bread.json [hunk 4]
```diff
@@ -400,6 +400,28 @@
     "fun": 1,
     "vitamins": [ [ "calcium", 1 ], [ "iron", 1 ] ]
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "tortilla_flour",
+    "name": { "str": "flour tortilla" },
+    "weight": "24 g",
```
Comment (Maleclypse) [inline data/json/items/comestibles/bread.json:407]:
```suggestion
    "weight": "18 g",
```

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Weights are just estimates of what I think might make the density tests pass.  Calorie checks should be fixed by these changes though. :)

**Comment by onura46** [comment]
> Weights are just estimates of what I think might make the density tests pass. Calorie checks should be fixed by these changes though. :)

Thank you so much for your suggestions! I was getting ready to reassess this and ask for details/docs about correcting item densities literally as you shared them <3

**Comment by onura46** [comment]
The item densities should be properly fixed now. Sorry for all the commits; I was just confused alternating with lazy trying to figure out the math I had to do, but turns out I was just reading the log incorrectly.

The default calories for flour tortillas still seem way too high, maybe twice as much, compared to reality. I'm guessing its just a byproduct of the ingredients and materials.

**Pull request merged**